### PR TITLE
move the gcc6 compatibility check to occur on a per-directory basis, …

### PR DESCRIPTION
**Pull request by zachlowry**
Proposed fix for #26233 https://github.com/opencv/opencv/issues/26233

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: cmake/OpenCVUtils.cmake [hunk 1]
```diff
@@ -353,23 +353,23 @@ function(ocv_target_include_directories target)
...
   foreach(dir ${ARGN})
     if("${dir}" STREQUAL "SYSTEM")
       set(__var_name __system_params)
     else()
-      get_filename_component(__abs_dir "${dir}" ABSOLUTE)
-      ocv_is_opencv_directory(__is_opencv_dir "${dir}")
-      if(__is_opencv_dir)
-        list(APPEND ${__var_name} "${__abs_dir}")
+      if(CV_GCC AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "6.0" AND
+          "${dir}" MATCHES "/usr/include$")
```
Comment (opencv-alalek) [inline cmake/OpenCVUtils.cmake:363]: > `"${dir}"`

Please use `dir MATCHES ...` without variable unpacking.

**Review by opencv-alalek — APPROVED** [review]

**Comment by asmorkalov** [comment]
@opencv-alalek

File: cmake/OpenCVUtils.cmake (hunk 1 above)
Comment (opencv-alalek) [reply cmake/OpenCVUtils.cmake]: Use `dir MATCHES "/usr/include$"`

See line 309

Comment (zachlowry) [reply cmake/OpenCVUtils.cmake]: got it! sorry about that

**Review by opencv-alalek — APPROVED** [review]
Thank you for contribution :+1:

**Pull request merged**